packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -1060,36 +1102,50 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
     ): Promise<CrawlerAddRequestsResult> {
         const requestQueue = await this.getRequestQueue();
 
-        if (!this.respectRobotsTxtFile) {
-            return requestQueue.addRequestsBatched(requests, options);
-        }
+        const requestLimit = this.calculateEnqueuedRequestLimit();
 
         const allowedRequests: (string | Source)[] = [];
-        const skipped = new Set<string>();
+        const skippedBecauseOfRobots = new Set<string>();
+        const skippedBecauseOfLimit = new Set<string>();
 
         for (const request of requests) {
             const url = typeof request === 'string' ? request : request.url!;
 
+            if (requestLimit !== undefined && allowedRequests.length >= requestLimit) {
+                skippedBecauseOfLimit.add(url);
+                continue;
+            }
+
             if (await this.isAllowedBasedOnRobotsTxtFile(url)) {
                 allowedRequests.push(request);
             } else {
-                skipped.add(url);
-                await this.onSkippedRequest?.({ url, reason: 'robotsTxt' });
+                skippedBecauseOfRobots.add(url);
+                await this.handleSkippedRequest({ url, reason: 'robotsTxt' });
             }
         }
 
-        if (skipped.size > 0) {
+        if (skippedBecauseOfRobots.size > 0) {
```

Comment:
Can we log this msg in `handleSkippedRequest`? No deeper idea behind this, just putting similar things together

Reply:
Ah, I see we can skip a request due to `robots.txt` even [once we start processing it](https://github.com/apify/crawlee/pull/2990/files#diff-3606a49356303bfa2bfed957bc81701e3d45d4cdafd4d5182685ef826c93b63fR1444)... it's a sticky situation, it feels like there is some space for refactor, but I cannot really put a finger on it. 

Feel free to ignore this.